app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt

```diff
@@ -0,0 +1,134 @@
+import com.karhoo.samples.uisdk.dropin.config.TokenExchangeConfig
+import com.karhoo.sdk.api.KarhooApi
+import com.karhoo.sdk.api.KarhooError
+import com.karhoo.sdk.api.network.request.UserLogin
+import com.karhoo.sdk.api.network.response.Resource
+import com.karhoo.uisdk.KarhooUISDK
+import com.karhoo.uisdk.screen.booking.BookingActivity
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var loadingProgressBar: View
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        KarhooApi.userService.logout()
+
+        loadingProgressBar = findViewById<View>(R.id.loadingSpinner)
+
+        findViewById<Button>(R.id.bookTripButtonGuest).setOnClickListener {
+            showLoading()
+
+            applyGuestConfig()
+
+            goToBooking()
+        }
+
+        findViewById<Button>(R.id.bookTripButtonTokenExchange).setOnClickListener {
+            showLoading()
+
+            applyTokenExchangeConfig()
+
+            loginTokenExchange()
+        }
+
+
+        findViewById<Button>(R.id.bookTripButtonLogin).setOnClickListener {
+            showLoginInputDialog()
+        }
+    }
+
+    private fun showLoginInputDialog() {
+        val dialog = Dialog(this)
+        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
+        dialog.setCancelable(false)
+        dialog.setContentView(R.layout.view_login)
+        val userNameEditText = dialog.findViewById(R.id.emailInput) as TextInputEditText
+        val passwordEditText = dialog.findViewById(R.id.passwordInput) as TextInputEditText
+        val signInButton = dialog.findViewById(R.id.signInButton) as Button
+        signInButton.setOnClickListener {
+            showLoading()
+            loginUser(userNameEditText.text.toString(), passwordEditText.text.toString())
+            dialog.dismiss()
+        }
+        dialog.show()
+    }
+
+    private fun loginUser(email: String, password: String) {
+        KarhooApi.userService.loginUser(UserLogin(email = email, password = password))
+                .execute { result ->
+                    when (result) {
+                        is Resource.Success -> goToBooking()
+                        is Resource.Failure -> toastErrorMessage(result.error)
+                    }
+                }
+    }
+
+    private fun applyTokenExchangeConfig() {
+        KarhooUISDK.apply {
+            setConfiguration(
+                TokenExchangeConfig(
+                    applicationContext
+                )
+            )
+        }
+    }
+
+    private fun applyGuestConfig() {
+        KarhooUISDK.apply {
+            setConfiguration(
+                GuestConfig(
+                    applicationContext
+                )
+            )
+        }
+    }
+
+    private fun loginTokenExchange() {
+        KarhooApi.userService.logout()
+        val token: String = BuildConfig.BRAINTREE_AUTH_TOKEN
```

Comment:
Needs to be able to handle Adyen auth token as well.

Reply:
Also the same for Guest User. Would be better to be able to switch easily between Adyen and Braintree.